zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt

```diff
@@ -0,0 +1,476 @@
+        mAngularSeekBar = binding.seekAngularSensitivity
+        mDeclareButton = binding.buttonDeclare
+        mCameraView = binding.imageViewCamera
+        mBatteryBar = binding.robotBatteryBar
+
+//        mCameraView.visibility = View.INVISIBLE
+
+
+        mDeclareButton.setOnClickListener(this)
+        mAngularSeekBar.setOnSeekBarChangeListener(this)
+        mLinearSeekBar.setOnSeekBarChangeListener(this)
+
+        mFwdButton.setOnTouchListener(this)
+        mBwdButton.setOnTouchListener(this)
+        mHomeButton.setOnClickListener(this)
+        mSoundButton.setOnClickListener(this)
+        mLeftButton.setOnTouchListener(this)
+        mRightButton.setOnTouchListener(this)
+
+
+        setScale()
+        declarePublisher()
+        startPubSub()
+
+        return root
+    }
+
+    private fun declarePublisher() {
+        mKeyExpr = mNSEditText.text.toString()+cmdVelSuffix
+        viewModel.zenohSession?.apply {
+            mKeyExpr.intoKeyExpr().onSuccess { ke ->
+                this.declarePublisher(ke).res().onSuccess { pub ->
+                    mPublisher = pub
+                    Log.v(TAG, "Declared publiser on: ${mNSEditText.text.toString()}")
+                }
+                    .onFailure { handleError(TAG, "Failed to launch publisher", it) }
+            }
+                .onFailure { handleError(TAG, "Failed to parse keyExpr", it) }
+        }
+    }
+
+
+    private fun publishCommand(cmd: Twist) {
+
+        var mStream = CDROutputStream()
+        cmd.serialize(mStream)
+        var arr = ByteArray(mStream.buffer.position())
+        mStream.buffer.rewind()
+        mStream.buffer.get(arr)
+        val payload = Value(arr, Encoding(KnownEncoding.APP_OCTET_STREAM))
+
+        mPublisher.put(payload).res().onFailure { Log.e(TAG, "Error when publishing", it) }
+            .onSuccess { Log.v(TAG, "Published command!") }
+    }
+
+    private fun setCmd(direction: MovementDirection) {
+        when (direction) {
+            MovementDirection.RIGHT -> {
+                mTwist.angular.z = (-1.0 * angularScale)
+            }
+
+            MovementDirection.LEFT -> {
+                mTwist.angular.z = (1.0 * angularScale)
+            }
+
+            MovementDirection.FWD -> {
+                mTwist.linear.x = (1.0 * linearScale)
+            }
+
+            MovementDirection.BWD -> {
+                mTwist.linear.x = (-1.0 * linearScale)
+            }
+
+            MovementDirection.STOP -> {
+                mTwist.linear.x = 0.0
+                mTwist.angular.z = 0.0
+            }
+        }
+        //publishCommand(mTwist)
+
+    }
+
+    private fun handleError(tag: String, errorMsg: String, error: Throwable) {
+        Log.e(tag, "$errorMsg: $error")
+        Toast.makeText(
+            activity,
+            errorMsg,
+            Toast.LENGTH_SHORT
+        ).show()
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
+            R.id.buttonDeclare -> {
```

Comment:
Here it'd be more efficient to do something like

```kotlin
mDeclareButton.setOnClickListener(v -> {
    declarePublisher()
    startPubSub()
})
```
and same thing for the other buttons